pkg/server/plugin/upstreamauthority/gcpcas/gcpcas.go

```diff
@@ -284,9 +286,16 @@ func (p *Plugin) mintX509CA(ctx context.Context, csr []byte, preferredTTL int32)
 	// after the path.Split call above.  We need to trim off the /certificateAuthorities/ part for the request below
 	chosenPool = strings.TrimSuffix(chosenPool, "/certificateAuthorities/")
 
+	// CertificateId is a unique identifier required for Enterprise tier CAs. We derive it
+	// from a hash of the public key, ensuring the same certificate request always yields the same resource name.
+	certIDHash := sha256.Sum256(csrParsed.RawSubjectPublicKeyInfo)
+	// Truncate the 32-byte hash to 28 bytes (56 hex characters) to comply with the limit: 7 ("spiffe-") + 56 = 63.
+	certificateID := "spiffe-" + hex.EncodeToString(certIDHash[:28])
```

Comment:
I'm not sure why we need this "spiffe-" prefix here. Keeping as much of the hash should make it easier for the id to be unique (although it's likely unique either way).

You could also use base32 instead of hex. I think either way it should be fine, but with base32 the sha256 hash will be encoded to less than 63 chars and will have the expected format. Either way, it might nicer to do the truncation at the end, maybe when adding the ID to the request object? `certificateID[:63]` makes it a bit easier to reason about than the `[:28]`, in my opinion.